blog/2022-12-13-dhis-to-rapidpro-in-the-field.md

```diff
@@ -0,0 +1,177 @@
+
+```shell
+export RAPIDPRO_API_TOKEN=98f3fe494b94742cf577f442e2cc175ae4f635a5
+
+./dhis2rapidpro.jar --rapidpro.api.url=https://rapidpro.dhis2.org/api/v2
+```
+
+Notice how the secret token used to access RapidPro was exported to an [environment variable](https://en.wikipedia.org/wiki/Environment_variable). For security reasons, secrets are not permitted in arguments passed to `dhis2rapidpro.jar`. We will see later on that you can place arguments and secrets in a config file read from `dhis2rapidpro.jar`.
+
+On the second run, we obtained another fatal error:
+
+>2022-11-14 20:18:45.914 ERROR 304446 --- [       	main] o.h.d.integration.rapidpro.Application   : TERMINATING!!! Missing DHIS2 API URL. Are you sure that you set \`dhis2.api.url\`?
+
+It was time to turn our attention to configuring access to DHIS2. DHIS-to-RapidPro supports two modes of authentication for DHIS2: [basic](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/introduction.html#webapi_basic_authentication) or [personal access token authentication](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/introduction.html#webapi_pat_authentication) (PAT). PAT was selected because it is considered more secure. We created a DHIS2 user dedicated to DHIS-to-RapidPro and generated a PAT from the user’s profile page that can GET and POST DHIS2 resources:
+
+![Personal access token dialog](./assets/dhis-to-rapidpro-in-the-field/pat.png)
+
+It is worth highlighting that the user was given permission to the organisation units (i.e., the villages) DHIS-to-RapidPro will be transmitting reports for. 
+
+The PAT, like the RapidPro API token, was exported to an environment variable while the `dhis2.api.url` argument was set to the [Web API](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/introduction.html) address of the MoHCC's DHIS2 instance:
+
+```shell
+export RAPIDPRO_API_TOKEN=98f3fe494b94742cf577f442e2cc175ae4f635a5
+export DHIS2_API_PAT=d2pat_apheulkR1x7ac8vr9vcxrFkXlgeRiFc94200032556
+
+./dhis2rapidpro.jar --rapidpro.api.url=https://rapidpro.dhis2.org/api/v2 \
+ --dhis2.api.url=https://play.dhis2.org/2.39.0/api
+```
+
+Third time's the charm and the application printed the banner saying it is operational along with the URLs to reach its various ancillary services:
+
+![First banner](./assets/dhis-to-rapidpro-in-the-field/banner-1.png)
+
+The ancillary services are available over HTTPS but, for the pilot, we decided to disable [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) given that this instance of DHIS-to-RapidPro sat behind a [reverse proxy server](https://www.cloudflare.com/learning/cdn/glossary/reverse-proxy/) that [terminated the client’s TLS connection](https://en.wikipedia.org/wiki/TLS_termination_proxy):
```

Comment:
Should a `:::caution` be added here to highlight SSL should never be turned off when it can be avoided, as it poses a security risk? It's easy to read over the note to not do it